torch/export/dynamic_shapes.py

```diff
@@ -53,6 +53,26 @@ class _DimHint(Enum):
     DYNAMIC = auto()
 
 
+class _DimHint:
+    AUTO = _DimHintType.AUTO
+    STATIC = _DimHintType.STATIC
+    DYNAMIC = _DimHintType.DYNAMIC
+
+    def __init__(self, _type, _min=None, _max=None):
```

Comment:
One nit I have with this design is that it's not clear that min and max are supposed to be "hints." E.g., hinting with `min=4` is compatible with the actual / effective `min` to be `8`, and values in this blind spot will result in runtime assertion errors. Do you have any ideas on how to address that?